common/events_io.hpp

```diff
@@ -85,6 +85,48 @@ class AsyncFileDescriptorWriter :
 #endif // MENDER_USE_BOOST_ASIO
 };
 
+class AsyncReaderFromReader : virtual public mender::common::io::AsyncReader {
```

Comment:
namespace mio = mender::common::io or something similar and shorter. Maybe just namespace common = mender::common at least, so that we get `common::io` as the namespace?